### ISSUE TITLE
NoneType' object has no attribute 'span' [BUG]

### ISSUE THREAD

**Issue by wkkskskdkkd** [pytube/pytube#1233]
**Before creating an issue**

Please confirm that you are on the latest version of pytube by installing from the source.
You can do this by running `python -m pip install git+https://github.com/pytube/pytube`.
Sometimes, the pypi library repository is not up to date, and your issue may have been fixed already!

**Describe the bug**

**To Reproduce**
Please provide the following information:
- The video or playlist url that is causing the error.
- The code where the problem is occurring.

**Expected behavior**

**Output**
If pytube raises an exception, please provide the full traceback for the exception.

**System information**
Please provide the following information:
- Python version (run `python --version`)
- Pytube version (run `print(pytube.__version__)` in python)
- Command used to install pytube

**Comment by Mettwasser** [comment]
Got the same issue

**Comment by glubsy** [comment]
Github is becoming a cesspool of idiots who can't read, can't write, can't understand anything. 

Don't act surprised when developers leave this platform if people keep spamming like this.

**Comment by xzh12** [comment]
cipher.py

**Comment by xzh12** [comment]
# name = re.escape(get_throttling_function_name(js))
    # print(js)
    name = "hha"

**Comment by ImranRahimov1995** [comment]
> # name = re.escape(get_throttling_function_name(js))
> ```
> # print(js)
> name = "hha"
> ```

it 's work thanks , i am really interested 

How do you know about  "hha"?

**Comment by MYtelecharger** [comment]
> # name = re.escape(get_throttling_function_name(js))
> ```
> # print(js)
> name = "hha"
> ```

Is it permanent fix or for only short time fix?

**Comment by NishantPacharne** [comment]
> # name = re.escape(get_throttling_function_name(js))
> ```
> # print(js)
> name = "hha"
> ```

This really worked man !! How the heck did you figured out this ?

**Comment by liamalon** [comment]
can you help, cant fix

**Comment by lucaArchidiacono** [comment]
statically assign "hha" as value to `name` should not be the correct way of handling this bug.
I extended the `get_throttling_function_name(js: str)` like following:
```
def get_throttling_function_name(js: str) -> str:
    """Extract the name of the function that computes the throttling parameter.

    :param str js:
        The contents of the base.js asset file.
    :rtype: str
    :returns:
        The name of the function used to compute the throttling parameter.
    """
    function_patterns = [
        # https://github.com/ytdl-org/youtube-dl/issues/29326#issuecomment-865985377
        # a.C&&(b=a.get("n"))&&(b=Dea(b),a.set("n",b))}};
        # In above case, `Dea` is the relevant function name
        r'a\.[A-Z]&&\(b=a\.get\("n"\)\)&&\(b=([^(]+)\(b\)',
    ]
    logger.debug('Finding throttling function name')
    for pattern in function_patterns:
        regex = re.compile(pattern)
        function_match = regex.search(js)
        if function_match:
            logger.debug("finished regex search, matched: %s", pattern)
            function_name = function_match.group(1)
            is_Array = True if '[' or ']' in function_name else False
            if is_Array:
                index = int(re.findall(r'\d+', function_name)[0])
                name = function_name.split('[')[0]
                pattern = r"var %s=\[(.*?)\];" % name
                regex = re.compile(pattern)
                return regex.search(js).group(1).split(',')[index]
            else:
                return function_name

    raise RegexMatchError(
        caller="get_throttling_function_name", pattern="multiple"
    )
```

The issue I guess is, the regex on line 269 is not expecting an array with it's according index.
I know this is not the most generic or best way of handling this kind of issue, but it gives an idea in which direction it should be handled.

**Comment by NishantPacharne** [comment]
> can you help, cant fix

A temporary fix for this be that you could comment the line no 293 name = re.escape(get_throttling_function_name(js)) in cipher.py and instead of that you could add name = "hha"

**Comment by lucaArchidiacono** [comment]
yes I know 😄  but to statically assign a value like "hha" only because it solves the current problem should not be way of handling bugs.
What if in the future instead of "hha" the value is "foo"? Then this would mean you need to adjust this line again with the new value. And this can go on and on...

**Comment by ghost** [comment]
name = "hha" no found

![image](https://user-images.githubusercontent.com/31140793/152563561-341394ce-5212-4bab-b29e-656ce91683b2.png)

**Comment by NishantPacharne** [comment]
> yes I know 😄 but to statically assign a value like "hha" only because it solves the current problem should not be way of handling bugs. What if in the future instead of "hha" the value is "foo"? Then this would mean you need to adjust this line again with the new value. And this can go on and on...

Yeah yeah ... True

**Comment by lucaArchidiacono** [comment]
I just found out that there is already an [open PR](https://github.com/pytube/pytube/pull/1222) from @glubsy which definitely works and looks way better than what I just wrote in here. @NishantPacharne check it out

**Comment by andipahlevy** [comment]
> statically assign "hha" as value to `name` should not be the correct way of handling this bug. I extended the `get_throttling_function_name(js: str)` like following:
> 
> ```
> def get_throttling_function_name(js: str) -> str:
>     """Extract the name of the function that computes the throttling parameter.
> 
>     :param str js:
>         The contents of the base.js asset file.
>     :rtype: str
>     :returns:
>         The name of the function used to compute the throttling parameter.
>     """
>     function_patterns = [
>         # https://github.com/ytdl-org/youtube-dl/issues/29326#issuecomment-865985377
>         # a.C&&(b=a.get("n"))&&(b=Dea(b),a.set("n",b))}};
>         # In above case, `Dea` is the relevant function name
>         r'a\.[A-Z]&&\(b=a\.get\("n"\)\)&&\(b=([^(]+)\(b\)',
>     ]
>     logger.debug('Finding throttling function name')
>     for pattern in function_patterns:
>         regex = re.compile(pattern)
>         function_match = regex.search(js)
>         if function_match:
>             logger.debug("finished regex search, matched: %s", pattern)
>             function_name = function_match.group(1)
>             is_Array = True if '[' or ']' in function_name else False
>             if is_Array:
>                 index = int(re.findall(r'\d+', function_name)[0])
>                 name = function_name.split('[')[0]
>                 pattern = r"var %s=\[(.*?)\];" % name
>                 regex = re.compile(pattern)
>                 return regex.search(js).group(1).split(',')[index]
>             else:
>                 return function_name
> 
>     raise RegexMatchError(
>         caller="get_throttling_function_name", pattern="multiple"
>     )
> ```
> 
> The issue I guess is, the regex on line 269 is not expecting an array with it's according index. I know this is not the most generic or best way of handling this kind of issue, but it gives an idea in which direction it should be handled.

thanks this is work for me

**Comment by xzh12** [comment]
index = js.index('function(a){var b=a.split("")')
    name = js[index-4:index-1]

**Comment by SaNcHeZ2222** [comment]
> index = js.index('function(a){var b=a.split("")')
> name = js[index-4:index-1]

it really works
